tests/lib/tools/filter-debug-output

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+
+import argparse
+import re
+import sys
+
+OPERATIONS = [
+    'Preparing', 'Executing', 'Restoring',
+    'Rebooting', 'Discarding', 'Allocating', 'Waiting',
+    'Allocated', 'Connecting', 'Connected', 'Sending',
+    'Error', 'Debug', 'WARNING'
+    ]
+
+def match_date(date):
+    return re.findall(r'\d{4}-\d{2}-\d{2}', date)
+
+def match_time(time):
+    return re.findall(r'\d{2}:\d{2}:\d{2}', time)
+
+# Debug line starts with 
+def match_debug(line):
+    parts = line.strip().split(' ')
+    return len(parts) > 5 and \
+        match_date(parts[0]) and \
+        match_time(parts[1]) and \
+        parts[2] == 'Debug' and \
+        parts[3] == 'output' and \
+        parts[4] == 'for'
+
+# Debug output finishes when a new other line starts
+def is_debug_finished(line):
+    parts = line.strip().split(' ')
+    return len(parts) > 3 and \
+        match_date(parts[0]) and \
+        match_time(parts[1]) and \
+        parts[2] in OPERATIONS
+
+
+# Process spread output lines
+def process_spread_output(debug_file):
+    debug_lines = ""
+    for line in sys.stdin:
```

Comment:
Could you enumerate the lines and print each line we actually print with a prefix that indicates the line number in the complete log. I think it might be easier to understand when cross-referencing the summary and the full log later on.

Reply:
The problem I see with this is that we will see 2 times the line number in the github logs

Also to keep the same identation (regarding the start of the log line) I need to add 0s to the line number dependending on the total number of lines, for example, when we have 6500 lines in the log I'll need to add three 0 to the first 10 lines, so, once the log is fully processsed I'll need iterate the lines and autocomplete which is not performance and makes difficult to show in real time the output.

Perhaps to make easy the cross-referencing, I can add a tag in both logs.